### PR TITLE
Allow Config::merge to accept a basic array.

### PR DESCRIPTION
**Pull request by dschissler**
I created a new clean PR with the alternate option that I presented in the last PR.  This will sit here for a few days until I have time to do the change log and to see about augmenting the tests.

ps. I wrote a metric shit load of zephir code today.

### CODE REVIEW

**Review by niden — CHANGES_REQUESTED** [review]
@dschissler This is solid. Can you please write a test and update the changelog?

**Comment by dschissler** [comment]
@niden Absolutely but I'll do it probably tomorrow.

File: phalcon/config.zep [hunk 1]
```diff
@@ -201,8 +201,21 @@ class Config implements \ArrayAccess, \Countable
...
+
+		switch typeof configParam {
+			case "array":
+				let config = new Config(configParam);
+				break;
+			case "object":
+				let config = configParam;
+				break;
+			default:
+				throw new \Exception("Invalid data type for merge.");
```
Comment (sergeyklay) [inline phalcon/config.zep]: Phalcon\Config\Exception

Comment (niden) [reply phalcon/config.zep]: Yes and this one! Sorry missed it

**Comment by dschissler** [comment]
So I wrote a test but I don't know what in the hell I'm doing.  I tried to run the tests on the command line but it failed out with other stuff.  Maybe it works, I have no idea.  It would be nice if there was something brain dead like `./run-tests`.

**Comment by dschissler** [comment]
@niden So it looks like new tests passed.  I wrote the logic correctly without being able to test it but then I forgot to add the stupid trait declaration and then the namespace use.  Do you want to take this or should I redo it to be less commits?  I still don't know how to run the tests from within my preferred VM.

**Review by niden — APPROVED** [review]

**Comment by niden** [comment]
Thank you @dschissler

**Comment by dschissler** [comment]
@niden np.  I have a question.  Do all of the commits get flattened into a single commit when you merge it?

**Comment by niden** [comment]
@dschissler As a general rule (self imposed) we use the squash and merge option when merging.

**Comment by dschissler** [comment]
@niden That is good to know.  Then I can just keep screwing with it until I get it right.

**Comment by niden** [comment]
@dschissler Honestly I still haven't sat down to read how to squash stuff on my local repo, so I do it here on github. ;)

**Pull request merged**